system-parachains/asset-hubs/asset-hub-polkadot/src/migration.rs

```diff
@@ -97,3 +106,221 @@ impl TryConvert<BlockNumberFor<Runtime>, BlockNumberFor<Runtime>> for RcToProxyD
+		match a {
+			RcPalletsOrigin::system(a) => Ok(OriginCaller::system(a)),
+			RcPalletsOrigin::Origins(a) => Ok(OriginCaller::Origins(a)),
+		}
+	}
+}
+
+/// Relay Chain Runtime Call.
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub enum RcRuntimeCall {
+	// TODO: variant set code for Relay Chain
+	// TODO: variant set code for Parachains
+	// TODO: whitelisted caller
+	#[codec(index = 19u8)]
+	Treasury(RcTreasuryCall),
+	#[codec(index = 21u8)]
+	Referenda(pallet_referenda::Call<Runtime>),
+	#[codec(index = 26u8)]
+	Utility(RcUtilityCall),
+	#[codec(index = 34u8)]
+	Bounties(pallet_bounties::Call<Runtime>),
+	#[codec(index = 38u8)]
+	ChildBounties(pallet_child_bounties::Call<Runtime>),
+}
+
+/// Relay Chain Treasury Call obtained from cargo expand.
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub enum RcTreasuryCall {
+	/// Propose and approve a spend of treasury funds.
+	#[codec(index = 3u8)]
+	spend_local {
+		#[codec(compact)]
+		amount: Balance,
+		beneficiary: Address,
+	},
+	/// Force a previously approved proposal to be removed from the approval queue.
+	#[codec(index = 4u8)]
+	remove_approval {
+		#[codec(compact)]
+		proposal_id: pallet_treasury::ProposalIndex,
+	},
+	/// Propose and approve a spend of treasury funds.
+	#[codec(index = 5u8)]
+	spend {
+		asset_kind: Box<VersionedLocatableAsset>,
+		#[codec(compact)]
+		amount: Balance,
+		beneficiary: VersionedLocation,
+		valid_from: Option<BlockNumber>,
+	},
+	/// Claim a spend.
+	#[codec(index = 6u8)]
+	payout { index: pallet_treasury::SpendIndex },
+	#[codec(index = 7u8)]
+	check_status { index: pallet_treasury::SpendIndex },
+	#[codec(index = 8u8)]
+	void_spend { index: pallet_treasury::SpendIndex },
+}
+
+/// Relay Chain Utility Call obtained from cargo expand.
+///
+/// The variants that are not generally used in Governance are not included.
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub enum RcUtilityCall {
+	/// Send a batch of dispatch calls.
+	#[codec(index = 0u8)]
+	batch { calls: Vec<RcRuntimeCall> },
+	/// Send a batch of dispatch calls and atomically execute them.
+	#[codec(index = 2u8)]
+	batch_all { calls: Vec<RcRuntimeCall> },
+	/// Dispatches a function call with a provided origin.
+	#[codec(index = 3u8)]
+	dispatch_as { as_origin: Box<RcPalletsOrigin>, call: Box<RcRuntimeCall> },
+	/// Send a batch of dispatch calls.
+	/// Unlike `batch`, it allows errors and won't interrupt.
+	#[codec(index = 4u8)]
+	force_batch { calls: Vec<RcRuntimeCall> },
+}
+
+/// Convert a Relay Chain Call to a local AH one.
+pub struct RcToAhCall;
+impl<'a> TryConvert<&'a [u8], RuntimeCall> for RcToAhCall {
+	fn try_convert(mut a: &'a [u8]) -> Result<RuntimeCall, &'a [u8]> {
+		let rc_call = match RcRuntimeCall::decode(&mut a) {
+			Ok(rc_call) => rc_call,
+			Err(e) => {
+				log::error!("Failed to decode RC call with error: {:?}", e);
+				return Err(a)
+			},
+		};
+		Self::map(rc_call).map_err(|_| a)
```

Comment:
Do you know what calls are in the referenda pallet that we do not translate? Should also be in the user impact

Reply:
not yet. my plan still to have everything translated. I think I will know later if we have anything we cannot translate